### PR TITLE
libtxt: search for fallback fonts that can match emoji and CJK characters

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Comment by jason-simmons** [comment]
@chinmaygarde

File: third_party/txt/src/txt/font_collection.cc [hunk 1]
```diff
@@ -30,6 +30,17 @@
 
 namespace txt {
 
+namespace {
+
+// Example characters representing character classes that may require a
```
Comment (gspencergoog) [inline third_party/txt/src/txt/font_collection.cc]: How are these examples?  Are they just characters that happen to be part of the character class, but you don't want/need to include the entire range?  When should we add characters to this list?

Comment (jason-simmons) [reply third_party/txt/src/txt/font_collection.cc]: Yes - these are characters which are part of a class that may not be covered by the platform's default font.

The Minikin font collection is constructed with a list of fonts and doesn't offer a way to search for additional fonts on demand in order to match a character.  This change attempts to build a list of fonts that contain some frequently seen character blocks and add them as backup options if the font collection's primary family doesn't handle a character.

The fallback_characters list should be extended if there is another character class that users expect to see rendered with the default text style but that is not contained in the platform default font.

File: third_party/txt/src/txt/font_collection.cc [hunk 2]
```diff
@@ -30,6 +30,17 @@
...
+// fallback font.
+const std::vector<SkUnichar> fallback_characters{
+    0x1f600,  // emoji
+    0x4e00,   // CJK
```
Comment (Hixie) [inline third_party/txt/src/txt/font_collection.cc]: the CJK case here won't be a long-term solution. What font to use when rendering CJK characters depends on the locale.

**Pull request merged**